controllers/permissions_controller.py

```diff
@@ -156,13 +156,20 @@ def index(self):
         for resource_type in query.all():
             resource_types[resource_type.name] = resource_type.description
 
+        # Create dict with all defined parents from resource objects
+        parents_dict = {}
+        for res in resources:
+            if res.resource.parent is not None and \
+                    res.resource.parent.id not in parents_dict.keys():
+                parents_dict[res.resource.parent.id] = res.resource.parent.name
+
         session.close()
 
         return render_template(
             '%s/index.html' % self.templates_dir, resources=resources,
-            endpoint_suffix=self.endpoint_suffix, pkey=self.resource_pkey(),
-            search_text=search_text, pagination=pagination,
-            sort=sort, sort_asc=sort_asc,
+            parents_dict=parents_dict, endpoint_suffix=self.endpoint_suffix,
+            pkey=self.resource_pkey(), search_text=search_text,
```

Comment:
Why not pass a method `parent_name` instead of a dict? Is it a performance problem?